Server/ReasnAPI/ReasnAPI/Services/EventService.cs

```diff
@@ -1,16 +1,255 @@
-﻿using ReasnAPI.Models.Database;
-
-namespace ReasnAPI.Services {
-    public class EventService (ReasnContext context) {
-        private readonly ReasnContext _context = context;
-
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
-    }
-}
+﻿using Microsoft.EntityFrameworkCore;
+using ReasnAPI.Exceptions;
+using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
+using System.Transactions;
+using ReasnAPI.Models.Enums;
+using ReasnAPI.Models.Mappers;
+using ReasnAPI.Exceptions;
+using System.Text.RegularExpressions;
+
+namespace ReasnAPI.Services;
+public class EventService(ReasnContext context)
+{
+    public EventDto CreateEvent(EventDto eventDto)
+    {
+        using (var scope = new TransactionScope())
+        {
+            
+            var createdTime = DateTime.UtcNow;
+            var newEvent = eventDto.ToEntity();
+            newEvent.CreatedAt = createdTime;
+            newEvent.UpdatedAt = createdTime;
+            newEvent.Slug = CreateSlug(eventDto, createdTime);
+           
+
+            context.Events.Add(newEvent);
+            context.SaveChanges();
+
+            var eventId = newEvent.Id;
+
+            if (eventDto.Tags is not null && eventDto.Tags.Any())
+            {
+                var newTags = eventDto.Tags
+                    .Where(t => !context.Tags.Any(x => x.Name == t.Name))
+                    .Select(t => new Tag { Name = t.Name })
+                    .ToList();
+
+                context.Tags.AddRange(newTags);
+                newEvent.Tags = newTags;

```

Comment:
What if tags are already on DB? Here you only attach newly created ones to event

Reply:
i changed it to add only new ones to db